overlord/snapstate/snapmgr.go

```diff
@@ -368,37 +361,21 @@ func (m *SnapManager) doDownloadSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	var storeInfo *snap.Info
-	if ss.UpdateInfo != nil {
-		storeInfo = &snap.Info{
-			AnonDownloadURL: ss.UpdateInfo.AnonDownloadURL,
-			DownloadURL:     ss.UpdateInfo.DownloadURL,
-			Confinement:     snap.ConfinementType(ss.UpdateInfo.Confinement),
-			SideInfo:        ss.UpdateInfo.SideInfo,
-		}
-	} else {
-		storeInfo, err = Store().Snap(ss.Name, ss.Channel, ss.DevMode(), auther)
```

Comment:
you still need a path doing this if DownloadInfo is not set to be backward compatible